### PR TITLE
Show "student" badge on student news

### PR DESCRIPTION
**Pull request by shagunvarma**
added a badge to news items that marks a news item to be for a "student" 
image

### CODE REVIEW

File: pages/news_items/news_items.ejs [hunk 1]
```diff
@@ -33,6 +33,9 @@
                     <% if (news_item.unread) { %>
                     <span class="badge badge-danger ml-2">Unread</span>
                     <% } %>
+                    <% if (news_item.visible_to_students == true) { %>
+                    <span class="badge badge-secondary ml-2">Student</span>
+                    <% } %>
```
Comment (mwest1066) [inline pages/news_items/news_items.ejs]: Can we add this badge only if the person viewing the issues page is not a student?

**Comment by mwest1066** [comment]
Can we also add the same page when viewing an individual news item?

Probably here: https://github.com/PrairieLearn/PrairieLearn/blob/master/pages/news_item/news_item.ejs#L26

**Comment by nicknytko** [comment]
Self assigning for my own selfish purposes &mdash; I would love for this to show up on the news articles for #2065.

**Comment by nicknytko** [comment]
image

**Comment by trombonekenny** [comment]
I'd probably drop that down to the "Posted on DATE" row and make it more explicit "Visible to students"?

**Comment by nicknytko** [comment]
> I'd probably drop that down to the "Posted on DATE" row and make it more explicit "Visible to students"?

Agreed, Dave, I think that looks much better.

image

File: pages/news_item/news_item.sql [hunk 2]
```diff
@@ -7,7 +7,8 @@ WITH mark_news_item_as_read AS (
 )
 SELECT
     ni.*,
-    format_date_only_no_tz(ni.date, coalesce(ci.display_timezone, c.display_timezone)) AS formatted_date
+    format_date_only_no_tz(ni.date, coalesce(ci.display_timezone, c.display_timezone)) AS formatted_date,
+    (users_is_course_staff($user_id) and ni.visible_to_students) AS show_student_badge
```
Comment (mwest1066) [inline pages/news_item/news_item.sql]: This is good, but I think we should change `users_is_course_staff()` to `STABLE` (currently it's `VOLATILE`). See https://www.postgresql.org/docs/12/xfunc-volatility.html for details. I believe it should be `STABLE`, and that would prevent it from being re-called on every row of this query.

File: pages/news_item/news_item.ejs [hunk 3]
```diff
@@ -15,7 +15,7 @@
       </a>
 
       <h1>
-        <%= news_item.title %>
+          <%= news_item.title %>
```
Comment (mwest1066) [inline pages/news_item/news_item.ejs]: Spurious whitespace change?

File: pages/news_item/news_item.ejs [hunk 4]
```diff
@@ -25,6 +25,9 @@
           by <%- news_item.author %>
           <% } %>
         </i>
+        <% if (news_item.show_student_badge) { %>
+            <span class="badge badge-secondary ml-2">Student Visible</span>
```
Comment (mwest1066) [inline pages/news_item/news_item.ejs]:
```suggestion
            <span class="badge badge-secondary ml-2">Visible to students</span>
```

File: pages/news_items/news_items.ejs [hunk 5]
```diff
@@ -33,6 +33,9 @@
                     <% if (news_item.unread) { %>
                     <span class="badge badge-danger ml-2">Unread</span>
                     <% } %>
+                    <% if (news_item.show_student_badge) { %>
+                    <span class="badge badge-secondary ml-2">Student</span>
```
Comment (mwest1066) [inline pages/news_items/news_items.ejs]:
```suggestion
                    <span class="badge badge-secondary ml-2">Visible to students</span>
```

**Comment by nicknytko** [comment]
@mwest1066 ping

File: pages/news_item/news_item.sql (hunk 2 above)
Comment (mwest1066) [inline pages/news_item/news_item.sql]:
```suggestion
    (users_is_course_staff($user_id) AND ni.visible_to_students) AS show_student_badge
```

File: pages/news_items/news_items.sql [hunk 6]
```diff
@@ -2,7 +2,8 @@
 SELECT
     ni.*,
     format_date_only_no_tz(ni.date, coalesce(ci.display_timezone, c.display_timezone)) AS formatted_date,
-    (an.id IS NOT NULL) AS unread
+    (an.id IS NOT NULL) AS unread,
+    (users_is_course_staff($user_id) and ni.visible_to_students) AS show_student_badge
```
Comment (mwest1066) [inline pages/news_items/news_items.sql]:
```suggestion
    (users_is_course_staff($user_id) AND ni.visible_to_students) AS show_student_badge
```

**Pull request merged**